rust/src/ortc.rs

```diff
@@ -641,29 +641,39 @@ pub(crate) fn can_consume(
 /// codecs' RTCP feedback and header extensions, and also enables or disabled RTX.
 #[allow(clippy::suspicious_operation_groupings)]
 pub(crate) fn get_consumer_rtp_parameters(
-    consumable_params: &RtpParameters,
-    caps: &RtpCapabilities,
+    consumable_rtp_parameters: &RtpParameters,
+    remote_rtp_capabilities: &RtpCapabilities,
     pipe: bool,
+    enable_rtx: bool,
 ) -> Result<RtpParameters, ConsumerRtpParametersError> {
     let mut consumer_params = RtpParameters {
-        rtcp: consumable_params.rtcp.clone(),
+        rtcp: consumable_rtp_parameters.rtcp.clone(),
         ..RtpParameters::default()
     };
 
-    for cap_codec in &caps.codecs {
+    for cap_codec in &remote_rtp_capabilities.codecs {
         validate_rtp_codec_capability(cap_codec)
             .map_err(ConsumerRtpParametersError::InvalidCapabilities)?;
     }
 
     let mut rtx_supported = false;
 
-    for mut codec in consumable_params.codecs.clone() {
-        if let Some(matched_cap_codec) = caps
+    for mut codec in consumable_rtp_parameters.codecs.clone() {
+        if !enable_rtx && codec.is_rtx() {
+            continue;
+        }
+
+        if let Some(matched_cap_codec) = remote_rtp_capabilities
             .codecs
             .iter()
             .find(|cap_codec| match_codecs(cap_codec.deref().into(), (&codec).into(), true).is_ok())
         {
-            *codec.rtcp_feedback_mut() = matched_cap_codec.rtcp_feedback().clone();
+            // *codec.rtcp_feedback_mut() = matched_cap_codec.rtcp_feedback().clone();
+
+            *codec.rtcp_feedback_mut() = matched_cap_codec
+                .rtcp_feedback().clone()
+                .retain(|fb| enable_rtx || fb != &RtcpFeedback::Nack);
```

Comment:
**TODO:** This is not compiling. Help needed.

Reply:
Resolved here but waiting for @nazar-pc's thoughts:
https://github.com/versatica/mediasoup/pull/1015/commits/a84ab48d923703f991d22f36fd6b22fda9992518

Reply:
@ibc you can use ".iter().filter(<condition>).collect()" without clone

Reply:
Yes, cloning just to move things around later makes less sense than just to filter stuff and collect then into a new collection instead. @Hartigan's solution is better. Fixed commit with corresponding change.

Reply:
Thanks guys